wallet.go

```diff
@@ -171,6 +176,40 @@ func (w *Wallet) updateNotificationLock() {
 	w.notificationLock = noopLocker{}
 }
 
+// CreditAccount returns the first account that can be associated
+// with the given credit.
+// If no account is found, ErrAccountNotFound is returned.
+func (w *Wallet) CreditAccount(c txstore.Credit) (uint32, error) {
+	_, addrs, _, _ := c.Addresses(activeNet.Params)
+	addr := addrs[0]
+	return w.Manager.AddrAccount(addr)
+}
+
+// DebitAccount returns the first account that can be associated
+// with the given debits.
+// If no account is found, ErrAccountNotFound is returned.
+func (w *Wallet) DebitAccount(d txstore.Debits) (uint32, error) {
+	msgTx := d.Tx().MsgTx()
+	for _, txIn := range msgTx.TxIn {
+		op := txIn.PreviousOutPoint
+		record, ok := w.TxRecord(&op.Hash)
+		if !ok {
+			return 0, btcjson.ErrNoTxInfo
+		}
+		for _, c := range record.Credits() {
```

Comment:
This is incorrect.  It should look up the account of credit that was spent, not the first credit created by the transaction (which may be sent to anther account's address).